src/client/powerpages/Contentsnippet.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import * as nls from "vscode-nls";
+nls.config({
+    messageFormat: nls.MessageFormat.bundle,
+    bundleFormat: nls.BundleFormat.standalone,
+})();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+import * as vscode from "vscode";
+import { fileName, folderName, isNullOrEmpty } from "./utils/CommonUtils";
+import { QuickPickItem } from "vscode";
+import { MultiStepInput } from "./utils/MultiStepInput";
+
+export const contentSnippet = async (context: vscode.ExtensionContext) => {
+    const { name, type } = await myMultiStepInput();
+
+    if (!isNullOrEmpty(name)) {
+        const terminal = vscode.window.createTerminal("Powerpages", "");
+        terminal.sendText(`cd data\n ../node_modules/.bin/yo @microsoft/powerpages:contentsnippet "${name}" "${type}"`);
+
+        const folder = folderName(name);
+        const file = fileName(name);
+
+        const watcher = vscode.workspace.createFileSystemWatcher(
+            new vscode.RelativePattern(
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                vscode.workspace.workspaceFolders![0],
+                `**/content-snippets/${folder}/${file}.en-US.contentsnippet.yml`
+            ),
+            false,
+            true,
+            true
+        );
+
+        context.subscriptions.push(watcher);
+
+        vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            cancellable: true,
+            title: localize("microsoft-powerapps-portals.webExtension.contentsnippet.create.message", "Creating Content Snippet..."),
+        }, async () => {
+            watcher.onDidCreate(async (uri) => {
+                await vscode.window.showTextDocument(uri);
+                terminal.dispose();
+            });
+        });
+    }
+};
+
+
+async function myMultiStepInput() {
+    const contentSnippetTypes: QuickPickItem[] = ["html", "text"].map(
+        (label) => ({ label })
+    );
+
+    interface State {
+        title: string;
+        step: number;
+        totalSteps: number;
+        type: QuickPickItem | string;
+        name: string;
+    }
```

Comment:
Looks like generic interface. Please make a separate file for this

Reply:
some properties are generic but as such it is specific to entities